app/controllers/items_controller.rb

```diff
@@ -0,0 +1,76 @@
+require 'net/http'
+
+class ItemsController < ApplicationController
+
+	def new
+		@item = Item.new
+	end
+
+	def create
+		@item = Item.new
+
+		url = item_params[:item_url]
+
+		if !validate_url(url)
+			flash[:error] = "Please enter a valid Ikea product URL or Article Number!"
+			render :new and return
+		end
+
+		create_item(@item, url, item_params)
+
+		if @item.save
+			flash[:notice] = "Here's your item!"
```

Comment:
I'd suggest acknowledging "successful" "creation" of the item somewhere in the flash message, even if you want to customize it a bit.